### PR TITLE
Update the build progress based on executor status

### PR DESCRIPTION
**Pull request by FenrirUnbound**
The status of the build remains `QUEUED` even after the build starts. This PR aims to accomplish two things:

1. Update the build status as `INPROGRESS` when the executor initiates the build.
1. Update the build status as `FAILURE` when the exectuor fails to initiate the build.

### CODE REVIEW

**Comment by stjohnjohnson** [comment]
Looks okay.

File: lib/buildmodel.js [hunk 1]
```diff
@@ -53,6 +52,47 @@ function fetchScmUrl(datastore, params, callback) {
...
+ * @method initializeBuildModel
+ * @param  {Object}             datastore         The datastore object that can retrieve the data from the datastore
+ * @param  {Object}             config            Config object of parameters
+ * @param  {String}             config.buildId    ID of the build
+ * @param  {String}             config.container  The kind of container being use
+ * @param  {String}             config.createTime Time that the build was created
+ * @param  {String}             config.jobId      ID of the related job
+ * @param  {String}             config.runNumber  The build run number
+ * @param  {Function}           callback  [description]
+ * @return {[type]}                       [description]
```
Comment (tkyi) [inline lib/buildmodel.js]: delete this

File: lib/buildmodel.js [hunk 2]
```diff
@@ -53,6 +52,56 @@ function fetchScmUrl(datastore, params, callback) {
     });
 }
 
+/**
+ * Callback for initializing the build model
+ *
+ * @callback initializeModelCallback
+ * @param {Error}   err              An error object, if an error is encountered
+ * @param {Object}  data             Data object
+ * @param {Object}  data.pipelineId  ID of the pipeline the build belongs to
```
Comment (d2lam) [inline lib/buildmodel.js]: String

File: lib/buildmodel.js [hunk 3]
```diff
@@ -53,6 +52,56 @@ function fetchScmUrl(datastore, params, callback) {
...
+ * @param {Object}  data.scmUrl      SCM URL to check out during the build
```
Comment (d2lam) [inline lib/buildmodel.js]: String also

File: lib/buildmodel.js [hunk 4]
```diff
@@ -53,6 +52,56 @@ function fetchScmUrl(datastore, params, callback) {
...
+ * the appropriate attributes from the related Pipeline and Job data
+ * @method initializeBuildModel
+ * @param  {Object}                   datastore         The datastore object that can retrieve the data from the datastore
+ * @param  {Object}                   config            Config object of parameters
+ * @param  {String}                   config.buildId    ID of the build
+ * @param  {String}                   config.container  The kind of container being use
+ * @param  {String}                   config.createTime Time that the build was created
+ * @param  {String}                   config.jobId      ID of the related job
+ * @param  {String}                   config.runNumber  The build run number
+ * @param  {initializeModelCallback}  callback          Callback to run when finished executing
```
Comment (d2lam) [inline lib/buildmodel.js]: createTime & runNumber are `number`

**Comment by nkatzman** [comment]
This seems fine 👍  Will want to refactor most of it inside of the different models anyway

**Comment by tkyi** [comment]
👍 Double comment is a little confusing but looks okay for first pass
https://github.com/screwdriver-cd/plugin-builds/pull/23/files#diff-2139c86722bcfb4545ab591a36fc73afR64

**Pull request merged**